File: LeetCode/Easy/0383-ransom-note/0383-ransom-note.go
func canConstruct(ransomNote string, magazine string) bool {
    if len(ransomNote) > len(magazine) {return false}
    counter := make([]int, 26)
    
    for _, v := range magazine {
        counter[v - 'a']++
    }
    
    location := 'a'
    for _, v := range ransomNote {
        location = v - 'a'
        if counter[location] < 1 { return false }
        counter[location]--
    }
    return true
}

File: LeetCode/Easy/0383-ransom-note/0383-ransom-note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        if (ransomNote.length() > magazine.length()) return false;
        int counter[26]{0};

        for (int i = 0; i < magazine.size(); ++i){
            ++counter[magazine[i] - 97];
        }
        int location;
        for (int i = 0; i < ransomNote.size(); ++i){
            location = ransomNote[i] - 97;
            if (counter[location] < 1) return false;
            counter[location]--;
        }
        return true;
    }
};